src/outputs/parthenon_hdf5.cpp

```diff
@@ -372,14 +464,102 @@ void PHDF5Output::WriteOutputFile(Mesh *pm, ParameterInput *pin, SimTime *tm) {
       WRITEH5SLAB("z", tmpData.data(), gLocations, local_start, local_count, global_count,
                   property_list);
     }
+    {
+      // open locations tab
+      H5G const gVLocations = H5G::FromHIDCheck(
+          H5Gcreate(file, "/VolumeLocations", H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT));
+
+      // write X coordinates
+      local_count[0] = num_blocks_local;
+      global_count[0] = max_blocks_global;
+
+      // These macros are defined in parthenon_hdf5.hpp, which establishes relevant scope
+      LOADVARIABLEALL(tmpData, pm, pmb->coords.x1v, out_ib.s, out_ib.e, 0, 0, 0, 0);
+      local_count[1] = global_count[1] = nx1;
+      WRITEH5SLAB("x", tmpData.data(), gVLocations, local_start, local_count,
+                  global_count, property_list);
+
+      // write Y coordinates
+      LOADVARIABLEALL(tmpData, pm, pmb->coords.x2v, 0, 0, out_jb.s, out_jb.e, 0, 0);
+      local_count[1] = global_count[1] = nx2;
+      WRITEH5SLAB("y", tmpData.data(), gVLocations, local_start, local_count,
+                  global_count, property_list);
+
+      // write Z coordinates
+      LOADVARIABLEALL(tmpData, pm, pmb->coords.x3v, 0, 0, 0, 0, out_kb.s, out_kb.e);
+
+      local_count[1] = global_count[1] = nx3;
+      WRITEH5SLAB("z", tmpData.data(), gVLocations, local_start, local_count,
+                  global_count, property_list);
+    }
+
+    {
+      // Write Levels and Logical Locations with the level for each Meshblock
+      // loclist contains levels and logical locations for all meshblocks on
+      // all ranks
+      const auto &loclist = pm->GetLocList();
+
+      std::vector<std::int64_t> levels;
+      levels.reserve(pm->nbtotal);
+
+      std::vector<std::int64_t> logicalLocations;
+      logicalLocations.reserve(pm->nbtotal * 3);
+
+      for (const auto &loc : loclist) { // for every block
+        levels.push_back(loc.level - pm->GetRootLevel());
+        logicalLocations.push_back(loc.lx1);
+        logicalLocations.push_back(loc.lx2);
+        logicalLocations.push_back(loc.lx3);
+      }
+
+      // Only write levels on rank 0
+      local_count[0] = (Globals::my_rank == 0) ? pm->nbtotal : 0;
+      local_count[1] = 0;
+      local_count[2] = 0;
+      local_count[3] = 0;
+      local_count[4] = 0;
```

Comment:
This is not necessary (same for `global_count`) since only the first entry will be ready when we create a rank 1 space